### PR TITLE
Cmake build

### PR DESCRIPTION
**Pull request by JonathanMeans**

### CODE REVIEW

File: modules/c/nitf/source/ImageIO.c [hunk 1]
```diff
@@ -6711,10 +6711,10 @@ NITFPRIV(int) nitf_ImageIO_writeMasks(_nitf_ImageIO * nitf,
 NITFPRIV(void) nitf_ImageIO_swapMaskHeader(_nitf_ImageIO_MaskHeader *
         header)
 {
-    nitf_Utils_byteSwap(&(header->imageDataOffset), 4);
```
Comment (adam-beauchamp) [inline modules/c/nitf/source/ImageIO.c]: Are these mods just to fix warnings?

File: modules/c/nrt/unittests/test_nrt_byte_swap.c [hunk 2]
```diff
@@ -31,7 +31,7 @@ TEST_CASE(testFourBytes)
 
     value = 0x12345678;
     swappedValue = 0x78563412;
-    nrt_Utils_byteSwap(&value, 4);
+    nrt_Utils_byteSwap((nrt_Uint8*)&value, 4);
```
Comment (adam-beauchamp) [inline modules/c/nrt/unittests/test_nrt_byte_swap.c]: Same question as above

File: modules/c/nitf/source/ImageIO.c (hunk 1 above)
Comment (JonathanMeans) [reply modules/c/nitf/source/ImageIO.c]: Yes

**Comment by hobu** [comment]
Lack of CMake was the only reason I was maintaining a fork at https://github.com/hobu/nitro

I am also the maintainer of the @conda-forge Nitro packages, and I'll be excited to update those to this main branch once the CMake branch is released.

**Comment by hobu** [comment]
OSX doesn't build at the moment 
```
n file included from ../modules/c++/sys/source/ScopedCPUAffinityUnix.cpp:32:
/Users/hobu/dev/git/nitro-main/externals/coda-oss/modules/c++/sys/include/sys/ScopedCPUAffinityUnix.h:72:11: error: unknown type name 'cpu_set_t'
    const cpu_set_t* getMask() const
          ^
/Users/hobu/dev/git/nitro-main/externals/coda-oss/modules/c++/sys/include/sys/ScopedCPUAffinityUnix.h:78:5: error: unknown type name 'cpu_set_t'
    cpu_set_t* getMask()
    ^
/Users/hobu/dev/git/nitro-main/externals/coda-oss/modules/c++/sys/include/sys/ScopedCPUAffinityUnix.h:97:5: error: unknown type name 'cpu_set_t'
    cpu_set_t* mMask;
    ^
../modules/c++/sys/source/ScopedCPUAffinityUnix.cpp:42:54: error: use of undeclared identifier 'CPU_SETSIZE'
    const int maxCPUs = std::max<int>(numOnlineCPUs, CPU_SETSIZE);
                                                     ^
... a bunch more
```

I would be willing to do some CI integration for OSX/Win64/Linux if you gave me some abilities to do so – installing Azure Pipelines would be a reasonable way to achieve that, and then I could tweak/edit the config in a PR until it was ready.

**Comment by JonathanMeans** [comment]
@hobu Thanks! I'm planning to configure various GitHub Actions and such later this week. I'll let you know when there's an Azure pipeline set up.

**Comment by JonathanMeans** [comment]
@adam-beauchamp Can you see https://github.com/mdaus/coda-oss/pull/346 as well?

**Pull request merged**